main.py

```diff
@@ -56,16 +57,31 @@ def pattern_match(patterns, source_list):
             task_names.add(matching)
     return sorted(list(task_names))
 
+def setup_example_logger(output_path, separator):
+    """Sets up a logger that will save each example and prediction."""
+    example_logger = logging.getLogger("examples")
+    filename = f"./outputs/examples{separator}{output_path}.jsonl"
+    formatter = logging.Formatter("%(message)s")
+    handler = logging.FileHandler(filename)
+    handler.setFormatter(formatter)
+    example_logger.addHandler(handler)
+    example_logger.setLevel(logging.INFO)
+
```

Comment:
We have `lm_eval/logger.py` where you can use `eval_logger.info` to log any messages. You can use that instead.

Reply:
I think I do not understand what you exactly mean.
To save the results of each example, I have used this logger. If we use `eval_logger`, it will also log other things which we do not find to get saved in examples json.
If you wanted me to add this example logger to `lm_eval/logger.py`, that's totally fine.

Reply:
Sorry, I meant if it's possible to make changes to `eval_logger` in `lm_eval/logger.py` to be able to do this instead of a new logger object?

Reply:
I think it is not possible to do that. Because `eval_logger` is logging several other things. In order to have a neat example logger, we had to define a new logger.

Reply:
Okay, sounds good then.